lib/cusparse/interfaces.jl

```diff
@@ -127,6 +127,77 @@ function Base.:(-)(A::CuSparseMatrix, B::CuSparseMatrixCSR)
     return geam(one(eltype(A)), csrA, -one(eltype(A)), B, 'O')
 end
 
+function Base.reshape(A::CuSparseMatrixCOO{T,M}, dims::NTuple{N,Int}) where {T,N,M}
+    nrows, ncols = size(A)
+    flat_indices = nrows * (A.colInd .- 1) + A.rowInd .- 1
+    new_col, new_row = div.(flat_indices, dims[1]) .+ 1, rem.(flat_indices, dims[1]) .+ 1
+    if length(dims) == 1
+        CuSparseMatrixCOO(sparse(new_row, new_col, A.nzVal, dims[1], 1))
+    else
+        CuSparseMatrixCOO(sparse(new_row, new_col, A.nzVal, dims[1], dims[2]))
+    end
+end
+
+function LinearAlgebra.mul!(Y::CuSparseMatrixCSR{T,M}, A::CuSparseMatrixCSR{T,M}, B::CuSparseMatrixCSR{T,M}) where {T,M}
+    CUSPARSE.version() < v"11.5.1" && throw(ErrorException("This operation is not supported by the current CUDA version."))
+    mm!('N', 'N', one(T), A, B, zero(T), Y, 'O')
+end
+function LinearAlgebra.mul!(Y::CuSparseMatrixCOO{T,M}, A::CuSparseMatrixCOO{T,M}, B::CuSparseMatrixCOO{T,M}) where {T,M}
+    Y2 = copy(CuSparseMatrixCSR(Y))
+    A2 = copy(CuSparseMatrixCSR(A))
+    B2 = copy(CuSparseMatrixCSR(B))
```

Comment:
Do we need `copy` here?
A new matrix is returned when we convert it.

Reply:
If I don't use the copy I get the error

```
UndefRefError: access to undefined reference

Stacktrace:
 [1] context
   @ ~/.julia/dev/CUDA/src/array.jl:281 [inlined]
 [2] resize!(A::CuArray{Int32, 1, CUDA.Mem.DeviceBuffer}, n::Int64)
   @ CUDA ~/.julia/dev/CUDA/src/array.jl:921
 [3] copyto!(dst::CuSparseMatrixCOO{Float64, Int32}, src::CuSparseMatrixCOO{Float64, Int32})
   @ CUDA.CUSPARSE ~/.julia/dev/CUDA/lib/cusparse/array.jl:480
 [4] mul!(Y::CuSparseMatrixCOO{Float64, Int32}, A::CuSparseMatrixCOO{Float64, Int32}, B::CuSparseMatrixCOO{Float64, Int32})
   @ Main ./In[22]:6
 [5] *(A::CuSparseMatrixCOO{Float64, Int32}, B::CuSparseMatrixCOO{Float64, Int32})
   @ CUDA.CUSPARSE ~/.julia/dev/CUDA/lib/cusparse/interfaces.jl:161
 [6] top-level scope
   @ In[24]:1
 [7] eval
   @ ./boot.jl:368 [inlined]
 [8] include_string(mapexpr::typeof(REPL.softscope), mod::Module, code::String, filename::String)
   @ Base ./loading.jl:1428
```

Reply:
That happens when you're working with an array whose storage has been freed (typically by calling `unsafe_free!`), and is indicative of a larger issue. You should not work around this by calling copy.

Reply:
Ok it should be fixed now.